org.alloytools.alloy.dash/src/main/java/ca/uwaterloo/watform/rapidDash/DashExprToPython.java

```diff
@@ -0,0 +1,310 @@
+                res = this.genExpr(node,1) + " is not None";
+                break;
+            case LONE:
+                res = " ";
+                break;
+            case ONE:
+                res = " ";
+                break;
+            case TRANSPOSE:
+                res = " ";
+                break;
+            case PRIME:
+                res = " ";
+                break;
+            case RCLOSURE:
+                res = " ";
+                break;
+            case CLOSURE:
+                res = " ";
+                break;
+            case CARDINALITY:
+                res = " ";
+                break;
+            case CAST2INT:
+                res = " ";
+                break;
+            case CAST2SIGINT:
+                res = " ";
+                break;
+            case NOOP:
+                res = "";
+                break;
+        }
+        return res;
+    }
+
+    // translate Binary operation, also returns the empty space
+    private String BinaryOp2PythonOp(ExprBinary.Op op){
+        String res = " ";
+        switch(op){
+            case ARROW:
+                res = " ";
+                break;
+            case ANY_ARROW_SOME:
+                res = " ";
+                break;
+            case ANY_ARROW_ONE:
+                res = " ";
+                break;
+            case ANY_ARROW_LONE:
+                res = " ";
+                break;
+            case SOME_ARROW_ANY:
+                res = " ";
+                break;
+            case SOME_ARROW_SOME:
+                res = " ";
+                break;
+            case SOME_ARROW_ONE:
+                res = " ";
+                break;
+            case SOME_ARROW_LONE:
+                res = " ";
+                break;
+            case ONE_ARROW_ANY:
+                res = " ";
+                break;
+            case ONE_ARROW_SOME:
+                res = " ";
+                break;
+            case ONE_ARROW_ONE:
+                res = " ";
+                break;
+            case ONE_ARROW_LONE:
+                res = " ";
+                break;
+            case LONE_ARROW_ANY:
+                res = " ";
+                break;
+            case LONE_ARROW_SOME:
+                res = " ";
+                break;
+            case LONE_ARROW_ONE:
+                res = " ";
+                break;
+            case LONE_ARROW_LONE:
+                res = " ";
+                break;
+            case ISSEQ_ARROW_LONE:
+                res = " ";
+                break;
+            case JOIN:
+                res = " ";
+                break;
+            case DOMAIN:
+                res = " ";
+                break;
+            case RANGE:
+                res = " ";
+                break;
+            case INTERSECT:
+                res = " ";
+                break;
+            case PLUSPLUS:
+                res = " ";
+                break;
+            case PLUS:
+                res = " + ";
```

Comment:
I feel like we need to decide how should we translate all "objects" in Dash. Like they are just sets of tuples in the Alloy/Dash world, but we were translating "one sig" as a singleton object when doing manual translation.
When building the translator, do we want to make everything as sets? Or do we want to make some customization, such as for "one sig" or "some sig"?
I am not sure if it can be translated into "+" directly, it depends on whether you are adding two integers or two sets. And for sets, it needs to be like "A.union(B)" for set union.

Reply:
Those are the work of translating signatures into Python. The translation of operations is kind of independent from that (not completely). A plus sign is "+" in python, that's why it's + here. Also, if you want, we can define the operation of + between two signatures in our python code as `union`. However, that really depends on how we translate signatures.

Reply:
How about for now let’s assume everything is a set? If we want to do some optimization later, we will do code changes all at once. The problem here is plus sign in Python is not a set operation, the set operation equivalent is union. Like the + in Alloy and + in Python have different meanings.

Reply:
Sure, without any override of operators, we can just use `|`